File: uniffi/src/message/groups/controller_idx_sigs.rs
use cesrox::error::ParsideResult;
use cesrox::Group;
use cesrox::Siger;
pub use cesrox::message::groups::{ControllerIdxSigs, ControllerIdxSig};

pub fn controller_idx_sig_create(siger: Siger) -> ControllerIdxSig {
    ControllerIdxSig::new(siger)
}

pub fn controller_idx_sigs_create(value: Vec<ControllerIdxSig>) -> ControllerIdxSigs {
    ControllerIdxSigs::new(value)
}

pub fn controller_idx_sigs_qb64(controller_idx_sigs: &ControllerIdxSigs) -> ParsideResult<String> {
    controller_idx_sigs.qb64()
}

pub fn controller_idx_sigs_qb64b(controller_idx_sigs: &ControllerIdxSigs) -> ParsideResult<Vec<u8>> {
    controller_idx_sigs.qb64b()
}

pub fn controller_idx_sigs_qb2(controller_idx_sigs: &ControllerIdxSigs) -> ParsideResult<Vec<u8>> {
    controller_idx_sigs.qb2()
}

File: wrappers/kotlin/lib/src/main/kotlin/cesrox/groups/ControllerIdxSigs.kt
package cesrox.groups

import cesrox.primitives.Siger
import uniffi.cesrox.*
import uniffi.cesrox.ControllerIdxSig as ControllerIdxSigNative
import uniffi.cesrox.ControllerIdxSigs as ControllerIdxSigsNative

class ControllerIdxSig {
    val value: ControllerIdxSigNative

    constructor(siger: Siger) {
        value = controllerIdxSigCreate(siger.value)
    }

    constructor(nativeValue: ControllerIdxSigNative) {
        value = nativeValue
    }

    val siger get() = Siger(value.siger)
}

class ControllerIdxSigs {
    val value: ControllerIdxSigsNative

    constructor(couples: List<ControllerIdxSig>) {
        value = controllerIdxSigsCreate(couples.map { it.value })
    }

    constructor(couples: ControllerIdxSigsNative) {
        value = couples
    }

    fun get(index: Int): ControllerIdxSig {
        return ControllerIdxSig(value.value[index])
    }

    fun qb2(): List<UByte> {
        return controllerIdxSigsQb2(value)
    }
    fun qb64b(): List<UByte> {
        return controllerIdxSigsQb64b(value)
    }
    fun qb64(): String {
        return controllerIdxSigsQb64(value)
    }
}